### PR TITLE
[TypeScript][Angular] Better support for "Accept", "Content-Type"

### PR DESCRIPTION
**Pull request by stoetti**
### Description of the PR

fix for issue https://github.com/swagger-api/swagger-codegen/issues/6006

checks if the produces or consumes list is null or empty and adds the corresponding headers (Accept, Content-Type), with the value being a list seperated by semicolon, according to http-spec.

### CODE REVIEW

**Comment by wing328** [comment]
@stoetti thanks for the PR. 

cc @TiFu  @taxpon @sebastianhaas @kenisteward @Vrolijkx

File: modules/swagger-codegen/src/main/resources/typescript-angular/api.service.mustache [hunk 1]
```diff
@@ -175,6 +175,11 @@ export class {{classname}} {
             '{{{mediaType}}}'{{#hasMore}},{{/hasMore}}
             {{/consumes}}
         ];
+
+        if ((consumes != null) && (consumes.length > 0)) {
+            headers.set('Content-Type', consumes.join(';'));
```
Comment (TiFu) [inline modules/swagger-codegen/src/main/resources/typescript-angular/api.service.mustache]: Just a little code style change: could you please remove the braces around `consumes != null` and `consumes.length > 0`? (same below in line 202). Thanks for the PR.

**Comment by sebastianhaas** [comment]
Hey! Thanks for your PR. I'm afraid though (as discussed in #6006) that it won't be possible just to set the respective `Accept` or `Response` headers according to what the spec says.

As @wing328 mentioned in the source thread:
> We need to implement SelectHeaderContentType (example) and SelectHeaderAccept (example) in ApiClient similar to what we've done for other langauges (e.g. Ruby, PHP, C#).

So, in other words, you need to add a little logic between the list of possible content types and the one that actually gets chosen, based on what we actually support. There is no point in allowing for xml when we only support json parsing, but with your code, as far as I can judge, this could easily happen with an API that reports only support for xml.

**Comment by stoetti** [comment]
Hey!
So what is needed is a logic that only adds json-related mediatypes to the list provided in the Accpet or Content-Type header?

In my opinion the examples @wing328 provides in the comment on the issue also allow xml-related headers to be added as long as there are no application/json entry in produces or consumes.

**Comment by wing328** [comment]
> In my opinion the examples @wing328 provides in the comment on the issue also allow xml-related headers to be added as long as there are no application/json entry in produces or consumes.

@stoetti that's correct. Just to be clear most generators do not support XML payload at the moment with 1 or 2 exceptions (e.g. Java Jersey 1.x or 2.x)

I'm happy to merge this fix into master while you're working on the enhancement to add functions in selecting accept and content-type. Does it sound good to you?

**Comment by stoetti** [comment]
No worries, I will take care of the enhancements as soon as I find the time (after family vacation) and if you want you can leave the pull request open until then.

**Comment by wing328** [comment]
Tests passed with updated TS samples via https://travis-ci.org/swagger-api/swagger-codegen/builds/278882925

**Comment by wing328** [comment]
@stoetti thanks for the PR, which has been merged into master.

Have a nice weekend.

**Comment by gupbeheer** [comment]
@wing328 The code of @stoetti seems to have been deleted in a merge after you merged his code to master. It is not in the current master anymore. Did I overlook something?

**Comment by wing328** [comment]
cc @macjohnny

**Comment by macjohnny** [comment]
@gupbeheer yes, the code was removed, as it caused other issues like https://github.com/swagger-api/swagger-codegen/issues/6588 and as pointed out in @bedag-moo's https://github.com/swagger-api/swagger-codegen/pull/6295#issuecomment-333179969 does not offer any technical enhancement.

**Comment by wing328** [comment]
@gupbeheer can you try the latest master to see if the TS Angular client works for you? If it's still an issue, we might need to add back the function in selecting accept, content-type and fix the bug with the function instead of removing it.

SNAPSHOT version of 2.3.0 (master) can be found in the README.

**Comment by gupbeheer** [comment]
@wing328 @macjohnny Thank you for the quick and clear reply. I tried it with the latest master, but it does not work for me. We use a rest service that uses the mime-types (access and content-types) for versioning. I like the way it was solved as was discussed in #6006. I made my own implementation based on that. I will make a PR so you can check it. I still need to check with the pet-store and the file-upload example. I will make the PR once I have done that.

**Comment by gupbeheer** [comment]
@wing328 @macjohnny the PR is done: #6751

**Pull request merged**